inst/include/cpp11/doubles.hpp

```diff
@@ -37,6 +37,12 @@ inline typename r_vector<double>::underlying_type* r_vector<double>::get_p(bool
   }
 }
 
+template <>
+inline typename r_vector<double>::underlying_type const* r_vector<double>::get_const_p(
+    bool is_altrep, SEXP data) {
+  return REAL_OR_NULL(data);
+}
```

Comment:
Using utilities like `REAL_OR_NULL()` gives ALTREP types that can cheaply return a const pointer to an underlying buffer a chance to do so (like ALTREP views)